ibm/data_source_ibm_atracker_targets_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccIBMAtrackerTargetsDataSourceBasic(t *testing.T) {
+	targetName := fmt.Sprintf("tf_name_%d", acctest.RandIntRange(10, 100))
+	targetTargetType := "cloud_object_storage"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMAtrackerTargetsDataSourceConfigBasic(targetName, targetTargetType),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.ibm_atracker_targets.atracker_targets", "id"),
+					resource.TestCheckResourceAttrSet("data.ibm_atracker_targets.atracker_targets", "name"),
+					resource.TestCheckResourceAttrSet("data.ibm_atracker_targets.atracker_targets", "targets.#"),
+					resource.TestCheckResourceAttr("data.ibm_atracker_targets.atracker_targets", "targets.0.name", targetName),
+					resource.TestCheckResourceAttr("data.ibm_atracker_targets.atracker_targets", "targets.0.target_type", targetTargetType),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMAtrackerTargetsDataSourceConfigBasic(targetName string, targetTargetType string) string {
+	return fmt.Sprintf(`
+		resource "ibm_atracker_target" "atracker_target" {
```

Comment:
I think you need to rewrite the testcase to provsion first the dependent resources(cos, buckets...) and then resource "ibm_atracker_target"

or else the prereq here is we should have cos and bucket provsioned and read those variables from provider
_test.go(via enviornmental variables)

Reply:
I think this will need to be a hand edit.